wgpu-core/src/device/mod.rs

```diff
@@ -1199,8 +1217,9 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
         let device = &device_guard[device_id];
 
         // If there is an equivalent BGL, just bump the refcount and return it.
-        // Warning: this isn't valid logic when `id_in` is provided.
-        {
+        // This is only applicable for identity filters that are generating new IDs,
+        // so their inputs are `PhantomData` of size 0.
+        if mem::size_of::<Input<G, id::BindGroupLayoutId>>() == 0 {
```

Comment:
Is there any cleaner way of doing this, this feels _really_ hacky

Reply:
I agree! but I wasn't able to properly detect this in a simple way.
since it's solving a real problem, and it's correct, let's proceed and follow up if necessary